File: api/src/main/java/structure/twitterapi/model/Post.java
package structure.twitterapi.model;

import javax.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity(name = "posts")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;
    @Column(name = "image_path")
    private String imagePath;
    @OneToMany(mappedBy = "post", cascade = CascadeType.REMOVE)
    private List<Like> likes;
    @ManyToOne
    private UserAccount user;
    private LocalDateTime dateCreating;
}

File: frontend/src/app/model/post.ts
import { Like } from './like';
import { UserAccount } from './userAccount';

export interface Post {
  id: number;
  userId: number;
  imagePath: string;
  description: string;
  likeIds: number[];
  likes: Like[];
  user: UserAccount;
  dateCreating: Date;
}
